### ISSUE TITLE
Choose light/dark theme

### ISSUE THREAD

**Issue by gjactat** [roslynpad/roslynpad#567]
Hello,

Although my Windows 11 is set to dark mode, I prefer to use Roslynpad in light mode.
Is there any way to choose the theme? 

Thank you for your handy software!

**Comment by aelij** [comment]
Yes, you can set "BuiltInTheme: Dark" in the RoslynPad.json file.

**Comment by gjactat** [comment]
Thanks !

**Comment by sjsepan3** [comment]
> Yes, you can set "BuiltInTheme: Dark" in the RoslynPad.json file.

Should  probably be 
`"BuiltInTheme": "Dark"`

Also, did not work for me w/ v20 on Linux (Linux Mint 22 Cinnamon) -- sys info below.

BTW, what keyword would I use for a VSCode theme, or will only BuiltIn work?

System:
  Kernel: 6.8.0-51-generic arch: x86_64 bits: 64 compiler: gcc v: 13.3.0 clocksource: tsc
  Desktop: Cinnamon v: 6.2.9 tk: GTK v: 3.24.41 wm: Muffin v: 6.2.0 vt: 7 dm: LightDM v: 1.30.0
    Distro: Linux Mint 22 Wilma base: Ubuntu 24.04 noble
Machine:
  Type: Desktop System: Dell product: OptiPlex 5040 v: N/A serial: <superuser required> Chassis:
    type: 3 serial: <superuser required>
  Mobo: Dell model: 0R790T v: A00 serial: <superuser required> part-nu: 06BA
    uuid: <superuser required> UEFI-[Legacy]: Dell v: 1.2.7 date: 01/15/2016
Battery:
  Device-1: hidpp_battery_0 model: Logitech M570 serial: <filter> charge: 40% rechargeable: yes
    status: discharging
CPU:
  Info: quad core model: Intel Core i5-6500 bits: 64 type: MCP smt: <unsupported> arch: Skylake-S
    rev: 3 cache: L1: 256 KiB L2: 1024 KiB L3: 6 MiB
  Speed (MHz): avg: 875 high: 900 min/max: 800/3600 cores: 1: 900 2: 900 3: 800 4: 900
    bogomips: 25599
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: NVIDIA GP108 [GeForce GT 1030] driver: nvidia v: 535.183.01 arch: Pascal pcie:
    speed: 8 GT/s lanes: 4 ports: active: none off: HDMI-A-1 empty: DVI-D-1 bus-ID: 01:00.0
    chip-ID: 10de:1d01 class-ID: 0300
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 driver: X: loaded: nouveau
    unloaded: fbdev,modesetting,vesa failed: nvidia gpu: nvidia,nvidia-nvswitch display-ID: :0
    screens: 1
  Screen-1: 0 s-res: 3840x2160 s-dpi: 162 s-size: 602x341mm (23.70x13.43") s-diag: 692mm (27.24")
  Monitor-1: HDMI-0 res: 3840x2160 hz: 60 dpi: 163 size: 600x340mm (23.62x13.39")
    diag: 690mm (27.15") modes: N/A
  API: EGL v: 1.5 hw: drv: nvidia platforms: device: 0 drv: nvidia device: 2 drv: swrast gbm:
    drv: kms_swrast surfaceless: drv: nvidia x11: drv: nvidia inactive: wayland,device-1
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 535.183.01 glx-v: 1.4
    direct-render: yes renderer: NVIDIA GeForce GT 1030/PCIe/SSE2
Audio:
  Device-1: Intel 100 Series/C230 Series Family HD Audio vendor: Dell driver: snd_hda_intel
    v: kernel bus-ID: 00:1f.3 chip-ID: 8086:a170 class-ID: 0403
  Device-2: NVIDIA GP108 High Definition Audio driver: snd_hda_intel v: kernel pcie:
    speed: 8 GT/s lanes: 4 bus-ID: 01:00.1 chip-ID: 10de:0fb8 class-ID: 0403
  API: ALSA v: k6.8.0-51-generic status: kernel-api
  Server-1: PipeWire v: 1.0.5 status: active with: 1: pipewire-pulse status: active
    2: wireplumber status: active 3: pipewire-alsa type: plugin
Network:
  Device-1: Intel Ethernet I219-V vendor: Dell driver: e1000e v: kernel port: N/A bus-ID: 00:1f.6
    chip-ID: 8086:15b8 class-ID: 0200
  IF: enp0s31f6 state: up speed: 1000 Mbps duplex: full mac: <filter>
  IF-ID-1: virbr0 state: down mac: <filter>
Drives:
  Local Storage: total: 1.82 TiB used: 932.26 GiB (49.9%)
  ID-1: /dev/sda vendor: SanDisk model: SD8SBAT-256G-1006 size: 238.47 GiB speed: 6.0 Gb/s
    tech: SSD serial: <filter> fw-rev: 1006 scheme: GPT
  ID-2: /dev/sdb vendor: Western Digital model: WDS100T2B0A-00SM50 size: 931.51 GiB
    speed: 6.0 Gb/s tech: SSD serial: <filter> fw-rev: 20WD scheme: MBR
  ID-3: /dev/sdd vendor: Toshiba model: External USB HDD size: 698.64 GiB type: USB rev: 2.0
    spd: 480 Mb/s lanes: 1 tech: N/A serial: <filter> scheme: MBR
Partition:
  ID-1: / size: 233.62 GiB used: 112.15 GiB (48.0%) fs: ext4 dev: /dev/sda3
  ID-2: /boot/efi size: 47.2 MiB used: 6.1 MiB (12.9%) fs: vfat dev: /dev/sda2
  ID-3: /home size: 915.82 GiB used: 674.2 GiB (73.6%) fs: ext4 dev: /dev/sdb1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 1001.2 MiB (48.9%) priority: -2 file: /swapfile
USB:
  Hub-1: 1-0:1 info: hi-speed hub with single TT ports: 16 rev: 2.0 speed: 480 Mb/s lanes: 1
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 1-5:2 info: 816820090226 type: mass storage driver: usb-storage interfaces: 1
    rev: 2.0 speed: 480 Mb/s lanes: 1 power: 500mA chip-ID: 3207:0300 class-ID: 0806 serial: <filter>
  Device-2: 1-6:3 info: Lite-On SK-9020 keyboard type: keyboard,HID driver: hid-generic,usbhid
    interfaces: 2 rev: 1.1 speed: 1.5 Mb/s lanes: 1 power: 100mA chip-ID: 04ca:004f class-ID: 0300
  Device-3: 1-7:4 info: Toshiba America External USB HDD type: mass storage driver: usb-storage
    interfaces: 1 rev: 2.0 speed: 480 Mb/s lanes: 1 power: 2mA chip-ID: 0480:a004 class-ID: 0806
    serial: <filter>
  Device-4: 1-9:5 info: Logitech Unifying Receiver type: keyboard,mouse,HID
    driver: logitech-djreceiver,usbhid interfaces: 3 rev: 2.0 speed: 12 Mb/s lanes: 1 power: 98mA
    chip-ID: 046d:c52b class-ID: 0300
  Hub-2: 2-0:1 info: super-speed hub ports: 10 rev: 3.0 speed: 5 Gb/s lanes: 1 chip-ID: 1d6b:0003
    class-ID: 0900
Sensors:
  System Temperatures: cpu: 31.0 C pch: 59.0 C mobo: N/A gpu: nvidia temp: 37 C
  Fan Speeds (rpm): N/A gpu: nvidia fan: 86%
Repos:
  Packages: pm: dpkg pkgs: 4602
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/gambas-team-gambas3-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/gambas-team-gambas3-jammy.gpg] https: //ppa.launchpadcontent.net/gambas-team/gambas3/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/gambas-team-gambas3-noble.list
    1: deb [signed-by=/etc/apt/keyrings/gambas-team-gambas3-noble.gpg] https: //ppa.launchpadcontent.net/gambas-team/gambas3/ubuntu noble main
  Active apt repos in: /etc/apt/sources.list.d/google-earth-pro.list
    1: deb [arch=amd64] http: //dl.google.com/linux/earth/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/jonaski-strawberry-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/jonaski-strawberry-jammy.gpg] https: //ppa.launchpadcontent.net/jonaski/strawberry/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/krupkat-code-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/krupkat-code-jammy.gpg] https: //ppa.launchpadcontent.net/krupkat/code/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/krupkat-code-noble.list
    1: deb [signed-by=/etc/apt/keyrings/krupkat-code-noble.gpg] https: //ppa.launchpadcontent.net/krupkat/code/ubuntu noble main
  Active apt repos in: /etc/apt/sources.list.d/lightzone-team-lightzone-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/lightzone-team-lightzone-jammy.gpg] https: //ppa.launchpadcontent.net/lightzone-team/lightzone/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/microsoft-prod.list
    1: deb [arch=amd64,arm64,armhf signed-by=/usr/share/keyrings/microsoft-prod.gpg] https: //packages.microsoft.com/ubuntu/24.04/prod noble main
  Active apt repos in: /etc/apt/sources.list.d/nala-sources.list
    1: deb http: //mirrors.advancedhosters.com/ubuntu/ noble main restricted universe multiverse
    2: deb http: //mirror.cs.jmu.edu/pub/ubuntu/ noble main restricted universe multiverse
    3: deb http: //mirror.us-ny2.kamatera.com/ubuntu/ noble main restricted universe multiverse
    4: deb https: //mirror.umd.edu/ubuntu/ noble main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com wilma main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu noble main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu noble-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu noble-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ noble-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/teejee2008-foss-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/teejee2008-foss-jammy.gpg] https: //ppa.launchpadcontent.net/teejee2008/foss/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/vivaldi.list
    1: deb [arch=amd64] https: //repo.vivaldi.com/stable/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/vscodium.list
    1: deb [signed-by=/usr/share/keyrings/vscodium-archive-keyring.gpg] https: //download.vscodium.com/debs vscodium main
  Active apt repos in: /etc/apt/sources.list.d/warpdotdev.list
    1: deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/warpdotdev.gpg] https: //releases.warp.dev/linux/deb stable main
Info:
  Memory: total: 16 GiB available: 15.54 GiB used: 3.98 GiB (25.6%)
  Processes: 271 Power: uptime: 6h 42m states: freeze,mem,disk suspend: deep wakeups: 0
    hibernate: platform Init: systemd v: 255 target: graphical (5) default: graphical
  Compilers: clang: 18.1.3 alt: 14/16 gcc: 13.3.0 alt: 11/12 Client: Unknown python client
    inxi: 3.3.34

**Comment by aelij** [comment]
The light and dark default VS Code themes are included in RP. You can also specify a path to a custom VS Code theme with the CustomThemePath and CustomThemeType (Light/Dark) settings.

It would be very hard for me to debug your system :) You can try debugging the app yourself. The relevant code is here: 

https://github.com/roslynpad/roslynpad/blob/40136740b8fc53f375a9626afddf9d333744abc3/src/RoslynPad.Common.UI/ViewModels/MainViewModel.cs#L110

https://github.com/roslynpad/roslynpad/blob/40136740b8fc53f375a9626afddf9d333744abc3/src/RoslynPad.Avalonia/MainWindow.axaml.cs#L72

**Comment by sjsepan3** [comment]
> The light and dark default VS Code themes are included in RP. You can also specify a path to a custom VS Code theme with the CustomThemePath and CustomThemeType (Light/Dark) settings.
> 

Thank You for this info. 

> It would be very hard for me to debug your system :) You can try debugging the app yourself. The relevant code is here:
> 

Most issue-reporting systems expect the user to provide some info about the context, which is what I did above.  Here is a little more...

When run from the command line, there are no errors.
I am running the file ~/.local/share/applications/RoslynPad.
In that location there was no file literally named RoslynPad.json as you suggested above to gjactat. There was, however, one named RoslynPad.runtimeconfig.json, which I modified with the key/value for the built-in theme. When that did not work I went back and created one literally named RoslynPad.json by copying RoslynPad.runtimeconfig.json. That had no effect either.

I could try the custom theme keys, but if the built-in one doesn't appear, I don't expect success with custom ones.

Thank You for your suggestion to debug the app, but without error messages, I'm not inclined. (Since this was tacked onto an already-closed issue, and I'm not going to open one to pursue it any further, there is nothing more to do here.) Thank You for your time.
